src/rgw/rgw_appmain.cc

```diff
@@ -583,7 +585,9 @@ void rgw::AppMain::shutdown(std::function<void(void)> finalize_async_signals)
 {
   if (env.driver->get_name() == "rados") {
     reloader.reset(); // stop the realm reloader
-    static_cast<rgw::sal::RadosLuaManager*>(env.lua.manager.get())->unwatch_reload(dpp);
+    if (g_conf().get_val<bool>("rgw_lua_enable"))
```

Comment:
we cannot truct the value of the conf option anywhere else than in the init function.
please check on the lua manager pointer as an indicator of whether you need to call `unwatch_reload()`

Reply:
> we cannot truct the value of the conf option anywhere else than in the init function. please check on the lua manager pointer as an indicator of whether you need to call `unwatch_reload()`

i tested this, able to access the conf flag and it does call `unwatch_reload` on stopping of rgw.
and see the following message in the logs
`2025-01-28T16:34:36.336+0000 7f0703ab2700 20 rgw lua package reloader: Started watching for reloads of  lua_package_allowlist with handle: 93942658766464
2025-01-28T16:37:16.707+0000 7f085dabf840 20 rgw main: Stopped watching for reloads of lua_package_allowlist with handle: 93942658766464
`

Reply:
it will, coz the config value change will not reflect in `g_conf().get_val<bool>("rgw_lua_enable")`.
the `g_conf` is loaded only during the start, so any changes in middle will not have affect.
so lua started at startup and stopped in middle without stop will still be true under `g_conf`
also tested this.
started vstart with lua enabled
disabled the lua with `ceph config set client.rgw.8000 rgw_lua_enable false`
stopped the rgw
the `unwatch is still called`